crates/sui-core/src/authority.rs

```diff
@@ -3663,6 +3682,32 @@ impl AuthorityState {
             .create_owner_index_if_empty(genesis_objects, &epoch_store)
             .expect("Error indexing genesis objects.");
 
+        if epoch_store
+            .protocol_config()
+            .enable_multi_epoch_transaction_expiration()
+            && epoch_store.epoch() > 0
+        {
+            use typed_store::Map;
+            let previous_epoch = epoch_store.epoch() - 1;
+            let start_key = (previous_epoch, TransactionDigest::ZERO);
+            let end_key = (previous_epoch + 1, TransactionDigest::ZERO);
+            let has_previous_epoch_data = store
+                .perpetual_tables
+                .executed_transaction_digests
+                .safe_range_iter(start_key..end_key)
+                .next()
+                .is_some();
+
+            if !has_previous_epoch_data {
+                panic!(
+                    "enable_multi_epoch_transaction_expiration is enabled but no transaction data found for previous epoch {}. \
+                    This indicates the node was restored using an old version of sui-tool that does not backfill the table. \
+                    Please restore from a snapshot using the latest version of sui-tool.",
+                    previous_epoch
+                );
+            }
```

Comment:
Thanks for covering this case 👍

Reply:
If there are any concerns about panic'ing in the event `executed_transaction_digests` is empty - we could always address this another way, eg. sui-tool writes a "version" marker to the db indicating a certain version of sui-tool has been used to restore the node.